examples/uprobe/main.go

```diff
@@ -0,0 +1,169 @@
+// This program demonstrates how to attach an eBPF program to a uretprobe.
+// The program will be attached to the 'readline' symbol in the binary '/bin/bash' and print out
+// the line which 'readline' functions returns to the caller.
+package main
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"unsafe"
+
+	ringbuffer "github.com/cilium/ebpf/perf"
+	goperf "github.com/elastic/go-perf"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang-11 UProbeExample ./bpf/uprobe_example.c -- -I../headers -O2
+
+const bashPath = "/bin/bash"
+const symbolName = "readline"
+
+type Event struct {
+	Pid  uint32
+	Line [80]byte
+}
+
+func main() {
+	stopper := make(chan os.Signal, 1)
+	signal.Notify(stopper, os.Interrupt, syscall.SIGTERM)
+
+	// Increase rlimit so the eBPF map and program can be loaded.
+	if err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &unix.Rlimit{
+		Cur: unix.RLIM_INFINITY,
+		Max: unix.RLIM_INFINITY,
+	}); err != nil {
+		log.Fatalf("failed to set temporary rlimit: %v", err)
+	}
+
+	specs, err := NewUProbeExampleSpecs()
+	if err != nil {
+		log.Fatalf("error while loading specs: %v", err)
+	}
+
+	objs, err := specs.Load(nil)
+	if err != nil {
+		log.Fatalf("error while loading objects: %v", err)
+	}
+
+	symbolAddress, err := getSymbolAddress(bashPath, symbolName)
+	if err != nil {
+		log.Fatalf("error while getting symbol address: %v", err)
+	}
+
+	efd, err := openUProbe(bashPath, symbolAddress, true, uint32(objs.ProgramUprobeBashReadline.FD()))
+	if err != nil {
+		log.Fatalf("create and attach UProbe: %v", err)
+
+	}
+	defer unix.Close(efd)
+
+	rd, err := ringbuffer.NewReader(objs.MapEvents, os.Getpagesize())
+	if err != nil {
+		log.Fatalf("error while createing ringbuffer reader: %v", err)
+	}
+	defer func() {
+		_ = rd.Close()
```

Comment:
If you wait for stopped here to close the reader you can get rid of the nested go routine.

```suggestion
		<-stopper
		_ = rd.Close()
```